app/models/user.rb

```diff
@@ -13,21 +13,27 @@ class User < ApplicationRecord
   # users table has encrypted password_digest column
   has_secure_password
   PASSWORD_VALIDATION_REGEX = /\A
-    (?=.{8,})           # Must contain 8 or more characters
-    (?=.*\d)            # Must contain a digit
-    (?=.*[a-z])         # Must contain a lower case character
-    (?=.*[A-Z])         # Must contain an upper case character
-    (?=.*[[:punct:]])   # Must contain a symbol
-    (?!.*(\s|\\|<|>))   # Must not contain whitespace, \, <, or >
-    .*                  # Matches everything (above lines cover validity)
+    (?=.{8,})             # Must contain 8 or more characters
+    (?=.*\d)              # Must contain a digit
+    (?=.*[a-z])           # Must contain a lower case character
+    (?=.*[A-Z])           # Must contain an upper case character
+    (?=.*[[:punct:]])     # Must contain a symbol
+    (?!.*(\s|\\|\/|<|>))  # Must not contain whitespace, \\, \/, <, or >
+    .*                    # Matches everything (above lines cover validity)
   /x # x means ignore whitespace in Regex creation, not actual formatting
   validates :password, presence: true,
             format: { with: PASSWORD_VALIDATION_REGEX }
 
-
   # name validation
-  validates :name, presence: true, length: { maximum: 50 }
-
+  NAME_VALIDATION_REGEX = /\A
+    (?=.{2,})               # Must contain 2 or more characters
+    (?=.*(\S))              # Must contain a non-whitespace character
+    (?!.*(\?|&|\\|\/|<|>))  # Must not contain ?, &, \\, \/, <, or >
+  /x # x means ignore whitespace in Regex creation, not actual formatting
```

Comment:
This should have a line like the regex for passwords that makes sure it matches everything (so add a .* and explanation)

Reply:
Fixed